components/suggest/src/schema.rs

```diff
@@ -149,15 +157,165 @@ impl ConnectionInitializer for SuggestConnectionInitializer {
         Ok(db.execute_batch(SQL)?)
     }
 
-    fn upgrade_from(&self, _db: &Transaction<'_>, version: u32) -> open_database::Result<()> {
+    fn upgrade_from(&self, tx: &Transaction<'_>, version: u32) -> open_database::Result<()> {
         match version {
             1..=15 => {
                 // Treat databases with these older schema versions as corrupt,
                 // so that they'll be replaced by a fresh, empty database with
                 // the current schema.
                 Err(open_database::Error::Corrupt)
             }
+            16 => {
+                tx.execute(
+                    "
+                    CREATE TABLE dismissed_suggestions (
+                        url_hash INTEGER PRIMARY KEY
+                    ) WITHOUT ROWID;",
+                    (),
+                )?;
+                Ok(())
+            }
             _ => Err(open_database::Error::IncompatibleVersion(version)),
         }
     }
 }
+
+/// Clears the database, removing all suggestions, icons, and metadata.
+pub fn clear_database(db: &Connection) -> rusqlite::Result<()> {
+    db.execute_batch(
+        "
+        DELETE FROM meta;
+        DELETE FROM suggestions;
+        DELETE FROM icons;
+        DELETE FROM yelp_subjects;
+        DELETE FROM yelp_modifiers;
+        DELETE FROM yelp_location_signs;
+        DELETE FROM yelp_custom_details;
+        ",
+    )
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use sql_support::open_database::test_utils::MigratedDatabaseFile;
+
+    const V16_SCHEMA: &str = r#"
```

Comment:
Is there a way to reuse the constant `SQL` to avoid a duplication? Looks like we wanted a snapshot of `SQL` for V16, would this be a one-off test just for V16, or will we also need the same for future versions?

Reply:
Yes it's a snapshot, so it needs to be duplicated.  I'm hoping that we can use the same sql as a basic test for all future migrations: take an empty v16 database, run all the migrations, and test that nothing fails.